src/ephy-bookmark.c

```diff
@@ -240,6 +330,59 @@ ephy_bookmark_get_title (EphyBookmark *bookmark)
 }
 
 void
+ephy_bookmark_set_id (EphyBookmark *self,
+                      const char   *id)
+{
+  g_return_if_fail (EPHY_IS_BOOKMARK (self));
+  g_return_if_fail (id != NULL);
+
+  g_free (self->id);
+  self->id = g_strdup (id);
+}
+
+const char *
+ephy_bookmark_get_id (EphyBookmark *self)
+{
+  g_return_val_if_fail (EPHY_IS_BOOKMARK (self), NULL);
+
+  return self->id;
+}
+
+void
+ephy_bookmark_set_modified (EphyBookmark *self,
+                            double        modified)
+{
+  g_return_if_fail (EPHY_IS_BOOKMARK (self));
+
+  self->modified = modified;
+}
+
+double
+ephy_bookmark_get_modified (EphyBookmark *self)
+{
+  g_return_val_if_fail (EPHY_IS_BOOKMARK (self), -1);
+
+  return self->modified;
+}
+
+void
+ephy_bookmark_set_uploaded (EphyBookmark *self,
```

Comment:
@mcatanzaro Also change `ephy_bookmark_is_uploaded` to `ephy_bookmark_get_is_uploaded`?